docs/azure/role-assignments.md

```diff
@@ -0,0 +1,261 @@
+
+var api = builder.AddProject<Projects.Api>("api")
+                 .WithRoleAssignments(search, SearchBuiltInRole.SearchIndexDataReader)
+                 .WithReference(search);
+```
+
+When you use the `WithRoleAssignments` method, it replaces the default role assignments with the specified ones. This method requires two parameters: the resource to which the role assignment applies and the built-in role to assign. In the preceding example, the `search` resource is assigned the <xref:Azure.Provisioning.Search.SearchBuiltInRole.SearchIndexDataReader?displayProperty=nameWithType> role.
+
+When you replace the default role assignments with the `SearchIndexDataReader` role, the API project is restricted to only reading data from the Azure Search resource. This ensures the API project can't write data to the Azure Search resource.
+
+For more information, see [Azure built-in roles](/azure/role-based-access-control/built-in-roles).
+
+## Built-in role assignment reference
+
+All built-in roles are defined within the <xref:Azure.Provisioning> namespaces and are included in the corresponding [📦 Azure.Provisioning.*](https://www.nuget.org/packages?q=Azure.Provisioning) NuGet packages. Each .NET Aspire Azure hosting integration automatically depends on the appropriate provisioning package. For more information, see [Infrastructure as code](integrations-overview.md#infrastructure-as-code).
+
+The following sections list the built-in roles for each Azure provisioning type that can be used as a parameter to the `WithRoleAssignments` API.
+
+### Azure App Configuration
+
+The provisioning resource type is <xref:Azure.Provisioning.AppConfiguration.AppConfigurationStore>, and the built-in roles are defined in the <xref:Azure.Provisioning.AppConfiguration.AppConfigurationBuiltInRole> struct. The built-in roles are:
+
+- <xref:Azure.Provisioning.AppConfiguration.AppConfigurationBuiltInRole.AppConfigurationDataOwner?displayProperty=nameWithType>
+- <xref:Azure.Provisioning.AppConfiguration.AppConfigurationBuiltInRole.AppConfigurationDataReader?displayProperty=nameWithType>
+
+<!--
+For more information, see [.NET Aspire Azure App Configuration integration](../configuration/azure-app-configuration-integration.md).
+-->
+
+### Azure App Container
+
+The provisioning resource type is <xref:Azure.Provisioning.AppContainers.ContainerApp>, and the built-in roles are defined in the <xref:Azure.Provisioning.AppContainers.AppContainersBuiltInRole> struct. The built-in roles are:
+
+- <xref:Azure.Provisioning.AppContainers.AppContainersBuiltInRole.Contributor?displayProperty=nameWithType>
+- <xref:Azure.Provisioning.AppContainers.AppContainersBuiltInRole.Owner?displayProperty=nameWithType>
+- <xref:Azure.Provisioning.AppContainers.AppContainersBuiltInRole.Reader?displayProperty=nameWithType>
+
+<!--
+For more information, see [.NET Aspire Azure App Container integration](../hosting/azure-app-container-integration.md).
+-->
+
+### Azure Application Insights
```

Comment:
We don't support App Insights role assignments. We actually don't grant any right now, even by default.

Reply:
Logged [Aspire.Hosting.Azure.ApplicationInsights should configure Microsoft Entra authentication (dotnet/aspire#8480)](https://github.com/dotnet/aspire/issues/8480) for this.